app/models/metric/common.rb

```diff
@@ -149,8 +149,9 @@ def v_derived_cpu_total_cores_used
   Vmdb::Deprecation.deprecate_methods(self, :v_derived_logical_cpus_used => :v_derived_cpu_total_cores_used)
 
   # Applies the given time profile to this metric record
+  # unless record already refer to some time profile (which were used for aggregation)
   def apply_time_profile(profile)
-    unless profile.ts_in_profile?(timestamp)
+    unless time_profile_id || profile.ts_in_profile?(timestamp)
```

Comment:
`unless` with `||` is mind melting.  Can you convert this to a regular `if`? `unless a || b` == `if !(a || b) == `if !a && !b`.  Alternately, you may want to switch this to 2 guard clauses which is generally easier to read.